src/ExcelDnaDoc/HtmlHelp.cs

```diff
@@ -40,28 +42,39 @@ public static void Create(string dnaPath, string helpSubfolder = "HelpContent",
 
             new MethodListView { Model = addin }.Publish();
 
-            foreach (var group in addin.Categories)
+            //Perhaps better to actually enumerate the values?
+            List<Task> tasks = new List<Task>(runAsync ? 5000 : 0);
+
+            foreach (var group in addin.Categories) 
             {
-                new CategoryView { Model = group }.Publish();
+                Action categoryAction = new Action(() => new CategoryView { Model = group }.Publish());
+                Run(categoryAction, tasks, runAsync);
 
                 foreach (FunctionModel function in group.Functions)
                 {
-                    new FunctionView { Model = function }.Publish();
+                    Action functionAction = new Action(() => new FunctionView { Model = function }.Publish());
+                    Run(functionAction, tasks, runAsync);
                 }
             }
 
             // create Excel Commands content
-
             if (addin.Commands.Count() != 0)
             {
                 new CommandListView { Model = addin }.Publish();
 
                 foreach (var command in addin.Commands)
                 {
-                    new CommandView { Model = command }.Publish();
+                    Action commandAction = new Action(() => new CommandView { Model = command }.Publish());
+                    Run(commandAction, tasks, runAsync);
                 }
             }
 
+            //Will be empty if async not enabled
+            foreach(Task task in tasks)
+            {
+                task.Wait();
+            }
```

Comment:
:customs: I'd suggest we use `Task.WaitAll` instead of a loop here and catch for `AggregateException`  so we can tell the user about errors in the different threads